File: API/Entities/Member.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{

    public required string Id { get; set; }
    public required string Gender { get; set; }
    public required string DateOfBirth { get; set; }
    public required string Created { get; set; }
    public required string LastActive { get; set; }
    public string? Description { get; set; }
    public required string City { get; set; }
    public required string Country { get; set; }
    public string? ImageUrl { get; set; }

    [JsonIgnore]
    public List<Photo> Photos { get; set; } = [];

    [JsonIgnore]
    [ForeignKey(nameof(Id))]
    public AppUser AppUser { get; set; } = null!;
}

File: client/src/types/member.ts
export type Member = {
  id: string
  displayName: string
  gender: string
  dateOfBirth: string
  created: string
  lastActive: string
  description?: string
  city: string
  country: string
  imageUrl?: string
}

export type Photo = {
  id: number
  publicUrl: any
  url: string
  memberId: string
}
